frontend/src/components/pages/RegistrantExperience/index.tsx

```diff
@@ -109,62 +166,169 @@ const RegistrantExperiencePage = (): React.ReactElement => {
           />
         );
       case RegistrantExperienceSteps.ReviewRegistrationPage:
-        return (
-          <ReviewRegistration
-            isChecked={sampleRegisterField}
-            toggleChecked={() => setSampleRegisterField(!sampleRegisterField)}
-            campName={camp?.name || ""}
-          />
-        );
+        if (camp) {
+          return (
+            <ReviewRegistration campers={registeringCampers} camp={camp} />
+          );
+        }
+        return <></>;
       default:
         throw new Error("unexpected page");
     }
   };
 
+  const isCurrentStepCompleted = (step: RegistrantExperienceSteps) => {
+    switch (step) {
+      case RegistrantExperienceSteps.PersonalInfoPage:
+        return isPersonalInfoFilled;
+      case RegistrantExperienceSteps.AdditionalInfoPage:
+        return isAdditionalInfoFilled;
+      case RegistrantExperienceSteps.WaiverPage:
+        return isWaiverFilled;
+      case RegistrantExperienceSteps.ReviewRegistrationPage:
+        return isReviewRegistrationFilled;
+      default:
+        return false;
+    }
+  };
+
   const handleStepNavigation = (stepsToMove: number) => {
     const desiredStep = currentStep + stepsToMove;
     if (RegistrantExperienceSteps[desiredStep]) {
       setCurrentStep(currentStep + stepsToMove);
     } else if (desiredStep < 0) {
       alert("PLACEHOLDER - go to initial registration");
+    } else if (camp) {
```

Comment:
I don't think I understand this condition. the first condition takes us to a defined page in the flow. The second takes us to the initial session selection page (i think?). Shouldn't this be a separate function called `handleCheckoutClick` or something which is conditionally called by the next/checkout button?